src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,364 @@
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/following?type=artist",
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    public String extractCurrentTrackUri(String playerData) {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(playerData);
+
+            if (jsonNode.has("item")) {
+                JsonNode item = jsonNode.get("item");
+                if (item.has("uri")) {
+                    return item.get("uri").asText();
+                }
+            }
+
+            return null;
+        } catch (IOException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @Override
+    public String getPlayerInfo(String accessToken) {
+        RestTemplate restTemplate = new RestTemplate();
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("Authorization", "Bearer " + accessToken);
+
+        HttpEntity<String> request = new HttpEntity<>(headers);
+
+        ResponseEntity<String> response = restTemplate.exchange(
+                "https://api.spotify.com/v1/me/player",
+                HttpMethod.GET,
+                request,
+                String.class
+        );
+
+        return response.getBody();
+    }
+
+    public String extractPlayerData(String playerData) {
+        ObjectMapper objectMapper = new ObjectMapper();
+        StringBuilder result = new StringBuilder();
+
+        try {
+            JsonNode root = objectMapper.readTree(playerData);
+
+            String playbackType = root.path("currently_playing_type").asText();
+
+            result.append("Playback Type: ").append(playbackType).append("\n");
+
+            if (playbackType.equals("track")) {
+                JsonNode trackNode = root.path("item");
+                if (trackNode.isMissingNode()) {
+                    throw new Exception("No track is currently playing.");
+                }
+
+                String artistName = trackNode.path("artists").get(0).path("name").asText();
+                String albumName = trackNode.path("album").path("name").asText();
+                String trackName = trackNode.path("name").asText();
+
+                result.append("Artist: ").append(artistName).append("\n");
+                result.append("Album: ").append(albumName).append("\n");
+                result.append("Track: ").append(trackName).append("\n");
+            } else {
+                result.append("Currently playing: ").append(playbackType).append("\n");
+            }
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "Error processing player data";
+        }
+
+        return result.toString();
+    }
+
+    @Override
+    public void playCurrentSong() {
+        RestTemplate restTemplate = new RestTemplate();
```

Comment:
same here, apply (DRY) don't repeat yourself https://www.plutora.com/blog/understanding-the-dry-dont-repeat-yourself-principle